static/app/components/scrollCarousel.tsx

```diff
@@ -0,0 +1,148 @@
+import {t} from 'sentry/locale';
+import {space, type ValidSize} from 'sentry/styles/space';
+import {useRefChildrenVisibility} from 'sentry/utils/useRefChildrenVisibility';
+
+interface ScrollCarouselProps {
+  children: React.ReactNode;
+  className?: string;
+  gap?: ValidSize;
+}
+
+/**
+ * This number determines what percentage of an element must be within the
+ * visible scroll region for it to be considered 'visible'. If it is visible
+ * but slightly off screen it will be skipped when scrolling
+ *
+ * For example, if set to 0.85, and 15% of the element is out of the scroll
+ * area to the right, pressing the right arrow will skip over scrolling to
+ * this element, and will scroll to the next invisible one.
+ */
+const DEFAULT_VISIBLE_RATIO = 0.85;
+
+/**
+ * This number determines how many items to jump when scrolling left or right
+ * when the arrow buttons are clicked
+ */
+const DEFAULT_JUMP_ITEM_COUNT = 2;
+
+export function ScrollCarousel({children, className, gap = 1}: ScrollCarouselProps) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const {visibility, childrenEls} = useRefChildrenVisibility({
+    ref,
+    visibleRatio: DEFAULT_VISIBLE_RATIO,
+  });
+
+  const isAtStart = visibility.at(0);
+  const isAtEnd = visibility.at(-1);
+
+  const scrollLeft = useCallback(() => {
+    const scrollIndex = visibility.findIndex(Boolean);
+    // Clamp the scroll index to the first visible item
+    const clampedIndex = Math.max(scrollIndex - DEFAULT_JUMP_ITEM_COUNT, 0);
+    childrenEls[clampedIndex]?.scrollIntoView({
+      behavior: 'smooth',
+      block: 'nearest',
+      inline: 'start',
+    });
+  }, [visibility, childrenEls]);
+
+  const scrollRight = useCallback(() => {
+    const scrollIndex = visibility.findLastIndex(Boolean);
+    // Clamp the scroll index to the last visible item
+    const clampedIndex = Math.min(
+      scrollIndex + DEFAULT_JUMP_ITEM_COUNT,
+      visibility.length - 1
+    );
+    childrenEls[clampedIndex]?.scrollIntoView({
+      behavior: 'smooth',
+      block: 'nearest',
+      inline: 'end',
+    });
+  }, [visibility, childrenEls]);
+
+  return (
+    <ScrollCarouselWrapper>
+      <ScrollContainer ref={ref} className={className} style={{gap: space(gap)}}>
+        {children}
+      </ScrollContainer>
+      {!isAtStart && <LeftMask />}
+      {!isAtEnd && <RightMask />}
+      {!isAtStart && (
+        <StyledArrowButton
+          onClick={scrollLeft}
+          style={{left: 0}}
+          aria-label={t('Scroll left')}
+          icon={<IconArrow direction="left" />}
+        />
+      )}
+      {!isAtEnd && (
+        <StyledArrowButton
+          onClick={scrollRight}
+          style={{right: 0}}
+          aria-label={t('Scroll right')}
+          icon={<IconArrow direction="right" />}
+        />
+      )}
+    </ScrollCarouselWrapper>
+  );
+}
+
+const ScrollCarouselWrapper = styled('div')`
+  position: relative;
+  overflow: hidden;
+`;
+
+const ScrollContainer = styled('div')`
```

Comment:
Feels pretty good as is, but do you think it's worth adding `scroll-snap-type`?

Reply:
i'm still playing around with that